OTA.md

```diff
@@ -0,0 +1,112 @@
+
+
+The `version` field is equivalent to a `tag` and can be any arbitrary string of up to 128 characters. 
+After successfully applying the new firmware, this `version` will be stored in NVS. 
+This values is compared to that of the latest machine config from app.viam.com and will trigger the update process.
+
+
+## Requirements
+
+- an esp32 WROVER-E with 8MB or more of flash memory
+- a partition table (ex `partitions.csv`) with `otadata`, `ota_0`, and `ota_1` partitions
+
+
+## Primer
+
+Consider firmware built with the following partition table:
+
+```
+# ESP-IDF Partition Table
+
+# Name,   Type, SubType, Offset,  Size, Flags
+# Note: if you have increased the bootloader size, make sure to update the offsets to avoid overlap
+nvs,	       data,	nvs,	  0x9000,	0x6000,
+otadata,       data,	ota,	  0xF000,	0x2000,
+phy_init,      data,	phy, 	  0x11000, 	0x1000,
+ota_0,	       app,	    ota_0,	  ,		    0x377000,
+ota_1,	       app,	    ota_1,	  ,		    0x377000,
+```
+
+The `otadata` partition contains information on which OTA partition to boot from and the states of the `ota_*` partitions.
+
+The terms 'firmware' or 'binary' can be a bit generic.
+In this section, we will refer to two types of binaries that can be built.
+1. a Merged Image
+2. an [Application (App) Image](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/app_image_format.html)
+
+> Note: in the absence of a `factory` partition `ota_0` fills the same initial role.
+
+## Full Build
+
+If a device is built with the above partition table, the `make build-esp32-bin` command creates a Merged Binary that includes
+- the bootloader
+- the partition table mapping (`partitions.csv`)
+- populated partitions (with partition headers) according to the mapping
+/
+The command `make flash-esp32-bin` writes this entire merged binary to the device's flash memory.
+
+This is the build workflow which must be used to:
+- flash a new device for the first time
+- update a device's partition table
+  - for example, make a device capable of OTA.
+
+**This is not the build that should be hosted at the `url` in the service config.**
+You can confirm this by using `ls -l** in your build directory to compare the size of the binary to your partition table.
+
+## OTA Build
+The `ota` build produces an app image (described above), which internally consists of:
+- the type-specific partition header, `esp_app_desc_t**
+- the application image that contains the program instructions
+
+**This app image is what you must host, see [Firmware Hosting Options](#firmware-hosting-options).**
+
+This build must be within the size limits of the smallest `ota_*` partition in a device's *current* partition table. 
+
+In this document's example, both the `ota_0` and `ota_1` partitions are ~3.4 MB. 
+The command underlying `make build-esp32-ota` takes the partition table as input so it **should** fail, additionally, target devices must be programmed to verify at runtime that the assertion holds.
```

Comment:
Have you tried this? Does it actually fail? Does `--target-app-partition` help?

Reply:
discussed offline. it does fail, but there is ambiguity (at least when not using this flag) as to which (possible all)  partitions of subtype `app` are being checked. 

ultimately, users of micro-rdk are most likely using the partition tables we provide and advanced users can find advanced uses/info in espressif official docs